File: pgm/ABC_Dart/ABC_423_A.dart
// import 'dart:collection';
import 'dart:io';
// import 'dart:math';

void main() {
  final [int X, int C] = stdin
      .readLineSync()!
      .split(' ')
      .map(int.parse)
      .toList();
  const int thousand = 1000;
  const int zero = 0;
  int withdrawal = zero;
  for (int i = thousand * (X ~/ thousand); zero < i; i -= thousand) {
    if (zero <= X - (i + C * (i ~/ thousand))) {
      withdrawal = i;
      break;
    }
  }
  print(withdrawal);
  exit(0);
}

File: pgm/ABC_C#/ABC_423_A.cs
using System;
// using System.Collections.Generic;
// using System.Globalization;
// using System.IO;
// using System.Linq;
// using System.Text;
// using System.Text.RegularExpressions;

class Program
{
    static void Main()
    {
        string[] arr = Console.ReadLine().Split(' ');
        const int one = 1;
        const int thousand = 1000;
        const int zero = 0;
        int X = int.Parse(arr[zero]);
        int C = int.Parse(arr[one]);
        Console.WriteLine(thousand * (X / (thousand + C)));
        Environment.Exit(0);
    }
}